### ISSUE TITLE
Improve handling of Vite configuration

### ISSUE THREAD

**Issue by galvez** [fastify/fastify-vite#60]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Standardize on `config.js` as the configuration file automatically loaded for `fastify-vite` applications, both for brevity and clear distinction from `vite.config.js`. Nest Vite options under a `vite` option from the main export.

### Motivation

1. Ensuring Vite config loading goes through the Fastify plugin allows for dynamic changes.

2. Naming it differently than `vite.config.js` prevents clashes and unwanted behavior if the user is unaware of `fastify-vite` being used in a project and tries to run commands through Vite's CLI.

**Comment by devmattrick** [comment]
`config.js` seems like it could conflict with unrelated config files. I think it'd be better to scope it how other tools like sveltekit (`sveltekit.config.js`) and Astro (`astro.config.js`) do it by just changing the part before the `.config.js`. `fastify-vite.config.js` would be a better choice here imo, even though it's a bit wordy.

Also it's worth noting that users may want to define their configs with other extensions, notably `.cjs` and `.mjs`. Potentially TypeScript extensions too, but I'm not sure how/ if TypeScript configs are supported.

**Comment by galvez** [comment]
Oh, this issue is tracking something that was done a long time.

Just for clarity, when I said `config.js`, I meant the internal `config.js` module.

@fastify/vite requires the presence of a `vite.config.{ts,js,cjs,mjs}`, even though not technically that's an opinionated measure to ensure Fastify and Vite configurations stay separate, leading to better maintainability and preserving the independence of the Vite application from Fastify.
